pkg/mtu/cell.go

```diff
@@ -103,11 +106,15 @@ func newForCell(lc cell.Lifecycle, p mtuParams, cc Config) (MTU, error) {
 
 			if configuredMTU == 0 {
 				mgr := &MTUManager{
-					mtuParams: p,
-					Config:    c,
+					mtuParams:     p,
+					Config:        c,
+					localNodeInit: make(chan struct{}, 1),
```

Comment:
I don't think a buffered channel is super useful here: whatever buffer size we choose, there will always be corner cases where the buffer might be full. Better to change the send in the observer job to take into account the `ctx` too:

```go
	select {
	case <-ctx.Done():
		return nil
	case m.localNodeInit <- struct{}{}:
	}
```

so that we won't block when we are shutting down.

Reply:
Agreed, the buffering doesn't make much sense. Though I expect to only ever push a message on it once, no repeatedly. 

Using the select here is indeed a good idea for proper shutdown, in case anything funny ever happens. I also added a line to nil out the channel after first use, so its clear its a one time only thing.